controlplane/src/core/services/AdmissionWebhookController.ts

```diff
@@ -50,10 +52,19 @@ export class AdmissionWebhookController {
     this.logger.debug({ url, path: '/validate-config', ...req }, 'Sending admission validate-config webhook request');
 
     try {
+      const headers: Record<string, string> = {};
+      if (this.graphAdmissionWebhookSecret) {
+        const dataString = JSON.stringify(req);
```

Comment:
I would expect that you hash the exact content you're sending as "data" on the request (line 66) but it seems like that's not the case? Although it might result in the same outcome in the current case, wouldn't it be better to hash the actual content of the request and not a JSON encoded representation of the payload which is passed to data?

Reply:
createHmac expects a string

Reply:
can you pass the exact same string to the request (data)

Reply:
You shouldnt need to parse a json response. The signature on the other hand you have to do it no matter what.